Clarify the Vault guide and add Vault ID examples (#55111)
ansible/ansible: docs/docsite/rst/user_guide/vault.rst

<assemble_module>` modules, the file will be placed at the destination on the target host decrypted
(assuming a valid vault password is supplied when running the play).

.. note::
    The advantages of file-level encryption are that it is easy to use and that password rotation is straightforward with :ref:`rekeying <rekeying_files>`.
    The drawback is that the contents of files are no longer easy to access and read. This may be problematic if it is a list of tasks (when encrypting a variables file, :ref:`best practice <best_practices_for_variables_and_vaults>` is to keep references to these variables in a non-encrypted file).

Variable-level encryption
^^^^^^^^^^^^^^^^^^^^^^^^^

> As of version 2.3, Ansible supports encrypting single values inside a YAML file, using the `!vault` tag to let YAML and Ansible know it uses special processing. This feature is covered in more details :ref:`below <encrypt_string_for_use_in_yaml>`.

Reviewer: Ansible 2.3?
Docs are versioned, so I think the version note can be removed

Author: Certainly, although that comes from the current documentation, I only added the "ref" bit in that line.

I don't know the exact reason why it's mentionned here although I'd imagine it's a sufficiently important feature to be specific about version history.

If that's the case maybe we can move the version notice to the #use-encrypt-string-to-create-encrypted-variables-to-embed-in-yaml section?

> As of version 2.3, Ansible supports encrypting single values inside a YAML file, using the `!vault` tag to let YAML and Ansible know it uses special processing. This feature is covered in more details :ref:`below <encrypt_string_for_use_in_yaml>`.

Reviewer 2: ```suggestion
Ansible also supports encrypting single values inside a YAML file, using the `!vault` tag to let YAML and Ansible know it uses special processing. This feature is covered in more detail :ref:`below <encrypt_string_for_use_in_yaml>`.
```

> *Available since Ansible 2.4*

Reviewer 2: In line with the comment from @gundalow about the docs being versioned, I suggest that we take this out.

```suggestion
```

> A vault ID is an identifier for one or more vault secrets. Since Ansible 2.4,

Reviewer 2: Thinking along the same lines:

```suggestion
A vault ID is an identifier for one or more vault secrets;
```

Ansible supports multiple vault passwords.

Vault IDs provide labels to distinguish between individual vault passwords.

To use vault IDs, you must provide an ID *label* of your choosing and a *source* to obtain its password (either ``prompt`` or a file path):

.. code-block:: bash

   --vault-id label@source

This switch is available for all Ansible commands that can interact with vaults: :ref:`ansible-vault`, :ref:`ansible-playbook`, etc.

Vault-encrypted content can specify which vault ID it was encrypted with.

> 

Reviewer 2: I think the two paragraphs below should also be amended.

It is not important the reader learns about the history of ansible-vault and Ansible. This documentation should be written for someone coming to it fresh, using the current version of Ansible.

I think these paragraphs should be replace by a short note at the end of the section, something like:

```
.. note:
    Older versions of Ansible, before 2.4, only supported using one single vault password at a time. 
```

Author: Hello,

Unless I've mis-githubbed it should be done.
